pkg/oc/cli/admin/migrate/scc/scc.go

```diff
@@ -0,0 +1,392 @@
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/origin/pkg/oc/cli/admin/migrate"
+	securityapi "github.com/openshift/origin/pkg/security/apis/security"
+	policy "k8s.io/api/policy/v1beta1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	policyv1beta1client "k8s.io/client-go/kubernetes/typed/policy/v1beta1"
+	rbacv1client "k8s.io/client-go/kubernetes/typed/rbac/v1"
+	rbacv1helper "k8s.io/kubernetes/pkg/apis/rbac/v1"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/genericclioptions/resource"
+)
+
+// clusterRoleAndBindingNamePrefix is a prefix that each auto-generated ClusterRole and ClusterRoleBinding have.
+// It simplifies their identification.
+const clusterRoleAndBindingNamePrefix = "psp:"
+
+// pspPriorityAnnotationKey is a name of the annotation that could be present on PodSecurityPolicy and specifies its priority.
+// It should be a number. An empty value means 0 priority. Higher values means that the policy has higher priority.
+// This is an equivalent of the SCC.Priority field but applies to PSP and it's required for seamless transition.
+const pspPriorityAnnotationKey = "security.openshift.io/psp-priority"
+
+// Before going v1, Sysctls only appeared in the Annotations field. Since the server can be of an older
+// version, we should make sure that the migration is done properly, meaning the annoations should also
+// be migrated
+const sysctlsPodSecurityPolicyAnnotationKey string = "security.alpha.kubernetes.io/sysctls"
+
+var (
+	internalMigrateSCCShort = "Converts SCCs to similar PSPs"
+	internalMigrateSCCLong  = templates.LongDesc(`
+		Converts SecurityContextConstraints to the equivalent PodSecurityPolicy objects.
+		For each SCC it also creates a ClusterRole with the similar name and ClusterRoleBinding
+		for every user or group that SCC has. To easy identify such auto-generated roles and bindings,
+		their names are prefixed by 'psp:' string.
+
+		NOTE: it does not modify anything by default.`)
+
+	internalMigrateSCCExample = templates.Examples(`
+		# Perform a dry-run of migrating all objects
+		%[1]s
+
+		# To actually perform the migration, the confirm flag must be appended
+		%[1]s --confirm`)
+)
+
+type migrateSCCOptions struct {
+	kubeRbacClient   *rbacv1client.RbacV1Client
+	kubePolicyClient *policyv1beta1client.PolicyV1beta1Client
+
+	cachedRoles        *rbacv1.RoleList
+	cachedClusterRoles *rbacv1.ClusterRoleList
+
+	migrate.ResourceOptions
+}
+
+func newCmdMigrateSCCOptions(out, errout io.Writer) *migrateSCCOptions {
+	return &migrateSCCOptions{
+		ResourceOptions: migrate.ResourceOptions{
+			Out:    out,
+			ErrOut: errout,
+
+			AllNamespaces: true,
+			Include:       []string{"securitycontextconstraints.security.openshift.io"},
+		},
+	}
+}
+
+func NewCmdMigrateSCC(name, fullName string, f kcmdutil.Factory, _ io.Reader, out, errout io.Writer) *cobra.Command {
```

Comment:
When https://github.com/openshift/origin/pull/20587 lands, all commands should accept `genericclioptions.IOStreams` as input.